pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -214,9 +227,32 @@ func (r *MachineSyncReconciler) reconcileCAPIMachinetoMAPIMachine(ctx context.Co
 }
 
 // reconcileMAPIMachinetoCAPIMachine a MAPI Machine to a CAPI Machine.
+//
+//nolint:funlen
 func (r *MachineSyncReconciler) reconcileMAPIMachinetoCAPIMachine(ctx context.Context, mapiMachine *machinev1beta1.Machine, capiMachine *capiv1beta1.Machine) (ctrl.Result, error) {
 	logger := log.FromContext(ctx)
 
+	newCAPIOwnerReferences, err := r.convertMAPIMachineOwnerReferencesToCAPI(ctx, mapiMachine)
+	if err != nil {
+		if condErr := r.applySynchronizedConditionWithPatch(ctx, mapiMachine, corev1.ConditionFalse, reasonFailedToConvertMAPIMachineToCAPI, err.Error(), nil); condErr != nil {
+			return ctrl.Result{}, utilerrors.NewAggregate([]error{err, condErr})
+		}
+
+		var fe *field.Error
+		if errors.As(err, &fe) || errors.Is(err, errUnsuportedOwnerKindForConversion) {
+			if fe.Detail == errUnsupportedCPMSOwnedMachineConversion.Error() {
+				logger.Info("Not converting %v Machine API machine. Conversion for machines owned by control plane machine set is currently not supported", mapiMachine.Name)
+				return ctrl.Result{}, nil
+			}
+
+			logger.Error(err, "unable to convert Machine API machine to Cluster API, unsupported owner reference in conversion")
+
+			return ctrl.Result{}, nil
+		}
+
+		return ctrl.Result{}, fmt.Errorf("failed to convert Machine API machine owner references to Cluster API: %w", err)
```

Comment:
I don't think we should ever error in case of failed owner reference conversion.
As we decided here.
https://github.com/openshift/cluster-capi-operator/pull/271#issuecomment-2754897624

Reply:
This will otherwise cause infinite reconcile loops when, for example, MAPI MachineSets are owned by  more than one owner reference.

See for exampl case 
```
	if len(mapiMachine.OwnerReferences) > 1 {
```
in `convertMAPIMachineOwnerReferencesToCAPI()`.

Reply:
This should now only retry when
	if err := r.Get(ctx, key, &capiMachineSet); err != nil {
fails.

The terminal field errors are handled above.